files/KoreBuild/msbuild/KoreBuild.RepoTasks.Sdk/Sdk/Sdk.targets

```diff
@@ -1,25 +1,40 @@
 <Project>
 
-  <PropertyGroup Condition="'$(MicrosoftBuildVersion)' == ''">
-    <!--
-      The dependency versions here don't matter as long as the version is binary compatible with the version of MSBuild and NuGet
-      already embedded in Microsoft.NET.Sdk. These are essentially reference assemblies.
+  <Choose>
+    <When Condition=" '$(MSBuildRuntimeType)' == 'Core' ">
 
-      Between MSBuild 15.6 and 15.8, the ref packages dropped .NET Standard 1.x suppport.
-      Use the netstandard2.0 packages if possible, otherwise fallback to 15.6.
-    -->
-    <MicrosoftBuildVersion Condition="$(TargetFramework.StartsWith('netstandard1'))">15.6.82</MicrosoftBuildVersion>
-    <MicrosoftBuildVersion Condition="'$(MicrosoftBuildVersion)' == ''">15.8.166</MicrosoftBuildVersion>
-  </PropertyGroup>
+      <PropertyGroup Condition="'$(MicrosoftBuildVersion)' == ''">
+        <!--
+          The dependency versions here don't matter as long as the version is binary compatible with the version of MSBuild and NuGet
+          already embedded in Microsoft.NET.Sdk. These are essentially reference assemblies.
 
-  <ItemGroup Condition="'$(DisableImplicitFrameworkReferences)' != 'true'">
-    <!-- set as private assets all since we don't need this in the publish output folder -->
-    <PackageReference Include="Microsoft.Build" Version="$(MicrosoftBuildVersion)" IsImplicitlyDefined="true" PrivateAssets="All" />
-    <PackageReference Include="Microsoft.Build.Framework" Version="$(MicrosoftBuildVersion)" IsImplicitlyDefined="true" PrivateAssets="All" />
-    <PackageReference Include="Microsoft.Build.Tasks.Core" Version="$(MicrosoftBuildVersion)" IsImplicitlyDefined="true" PrivateAssets="All" />
-    <PackageReference Include="Microsoft.Build.Utilities.Core" Version="$(MicrosoftBuildVersion)" IsImplicitlyDefined="true" PrivateAssets="All" />
-    <!-- because almost everyone ends up using JSON in repo tasks -->
-    <PackageReference Include="Newtonsoft.Json" Version="$(JsonInMSBuildVersion)" IsImplicitlyDefined="true" PrivateAssets="All" />
-  </ItemGroup>
+          Between MSBuild 15.6 and 15.8, the ref packages dropped .NET Standard 1.x suppport.
+          Use the netstandard2.0 packages if possible, otherwise fallback to 15.6.
+        -->
+        <MicrosoftBuildVersion Condition="$(TargetFramework.StartsWith('netstandard1'))">15.6.82</MicrosoftBuildVersion>
+        <MicrosoftBuildVersion Condition="'$(MicrosoftBuildVersion)' == ''">15.8.166</MicrosoftBuildVersion>
```

Comment:
These two versions are, like, super specific. Is this fragile?

Reply:
There are two versions here because the MSBuild made breaking changes between 15.6 and 15.8. 15.6 only supports netstandard1.x, 15.8 only supports netstandard2.0. We prefer to use netstandard2.0 where possible (faster restores, fewer dependencies), but I kept the fallback to 15.6 for compatibility. In the master branch, I plan to remove netstandard1.x support.

Reply:
OK, but it's no problem that these are super exact versions? Like what if instead of 15.8.166 it's really 15.8.167? Is that even a thing?

Reply:
Right, not a problem. The very-specific versions are due to the way MSBuild versions the packages they release to nuget.org.

![image](https://user-images.githubusercontent.com/2696087/48432838-5fff2700-e72a-11e8-895e-b8e4003141d6.png)